xla/stream_executor/gpu/gpu_timer.cc

```diff
@@ -130,7 +130,11 @@ DeviceMemory<GpuSemaphoreState> GpuTimer::GpuSemaphore::device() {
     // Check the assumption that this device supports unified addressing,
     // otherwise skip the delay kernel
     TF_ASSIGN_OR_RETURN(int status, GpuDriver::GetDeviceAttribute(
+#if GOOGLE_CUDA
```

Comment:
This seems unrelated to the central point of this PR.  It doesn't seem related to the logging API.

In addition, it's sprinkling #ifdefs throughout the code -- as a general rule, ifdefs should be avoided in this manner.  It would be far preferable to refactor the GpuTimer to be ROCm specific if there need to be differences between CUDA and ROCm.

Reply:
It is unrelated. It is a build fix. Unfortunately, since you guys don't require PRs to build (let alone pass unit tests) on ROCm before accepting them, the trunk of OpenXLA tends to be broken on ROCm much of the time, and I have a choice between including all build fixes in my PRs, or having them in the same broken state as the trunk (in which case, I can't even run unit tests on them).

The discussion about ifdefs would probably be better directed at the person who originally submitted the CUDA specific code into the repository, breaking the AMDGPU build (https://github.com/openxla/xla/blame/main/xla/stream_executor/gpu/gpu_timer.cc#L133).

Reply:
If there's a build break, I would encourage you to file a separate PR to fix that build break before trying to submit an unrelated PR if that is causing you problems.  Adding multiple unrelated changes to a single PR can be problematic to effectively land.

As far as ROCm builds, I believe we are working with Chao Chen trying to get a ROCm build as part of openxla.

I can't change what was done in the past, but I can try and work with new PRs to try and fix things that were probably done incorrectly in the past.  I agree that having CUDA-specific code that breaks ROCm builds isn't great.  Fixing those build breaks is definitely something that should be done, but we should endeavor to fix them in the least brittle way to ensure it's easier to prevent those things from happening in the future.

Reply:
Hi @klucke  we already have a fixed PR on this build break, could you review and merge it, please? Thanks https://github.com/openxla/xla/pull/11205